File: Divide Array in Sets of K Consecutive Numbers.py
class Solution:
    def isPossibleDivide(self, nums: List[int], k: int) -> bool:
        counter = Counter(nums)
        while counter:
            smallest = min(counter.keys())
            for num in range(smallest, k+smallest):
                if num not in counter:
                    return False
                counter[num] -= 1
                if counter[num] == 0:
                    del counter[num]
        return True

File: Divide Array in Sets of K Consecutive Numbers.go
func isPossibleDivide(nums []int, k int) bool {
    counter := make(map[int]int)
    for _, num := range nums {
        if _, ok := counter[num]; !ok {
            counter[num] = 1
        } else {
            counter[num]++
        }
    }
    for len(counter) > 0 {
        smallest := -1
        for num := range counter {
            if smallest == -1 {
                smallest = num
            } else {
                smallest = min(smallest, num)
            }
        }

        for num := smallest; num < smallest+k; num++ {
            if _, ok := counter[num]; !ok {
                return false
            }

            counter[num]--
            if counter[num] == 0 {
                delete(counter, num)
            }
        }
    }
    return true
}
